morepath/app.py

```diff
@@ -255,17 +247,176 @@ def set_setting_section(section, section_settings):
         for section, section_settings in settings.items():
             set_setting_section(section, section_settings)
 
-    def _get_class_path(self, model, variables):
-        """Path for a model class and variables.
+    @reg.dispatch_method(get_key_lookup=cached_key_lookup)
+    def _view(self, obj, request):
```

Comment:
This method replaces ``generic.view``, which is discussed in ``view.rst`` as part of the User Guide, and is the only of the dispatched method for which it makes sense and fulfils a legitimate use-case to have it as an argument to ``@App.predicate``.  As such it should be public, i.e., no leading underscore.

As for the other dispatched methods, some should definitely be private (e.g., those related to path handling) others could be either, but I'm fine leaving them as private.